app/scripts/controllers/transactions/index.js

```diff
@@ -958,6 +960,7 @@ export default class TransactionController extends EventEmitter {
           txMeta.txParams.from,
           txMeta.destinationTokenDecimals,
           approvalTxMeta,
+          chainId,
```

Comment:
txMeta should now include the chainId for the transaction. I think this is safer than getting the current chainId if in the event that the network switches while this flow is in flight.

Reply:
addressed in 0a89657cc